public/app/features/dashboard/dashgrid/PanelChrome.tsx

```diff
@@ -78,6 +79,10 @@ export class PanelChrome extends PureComponent<Props, State> {
     });
   };
 
+  onInterpolate = (value: string, format?: string) => {
```

Comment:
I'm not 100% sure whether or not the interpolation should belong to `PanelChrome`. My gut feeling says interpolation is specific to a certain panel type and visualization, and should be implemented in `GaugePanel` in this case. Am I missing sth here @torkelo ?

Reply:
Good question, me and @torkelo did some investigation and we found this to be the best place. We could also move it even further out in the hierarchy.

Reply:
@dprokop string interpolation means interpolating a string that contains Grafana dashboard variables. Variables can either have global values (ie dashboard level) or scoped values local to the panel (for repeated panels). So in order not to have panels worry about scoped variables we decided to hand them a function that is scoped to this panel and handles the interpolation. 

The GaugePanel then use this function to interpolate the prefix & suffix props that can contain dashboard variables, before handing the props to the Gauge visualization 
https://github.com/grafana/grafana/blob/master/public/app/plugins/panel/gauge/GaugePanel.tsx#L13